sdk/identity/identity/samples/ClientSideUserAuthentication.md

```diff
@@ -0,0 +1,62 @@
+
+```ts
+const { InteractiveBrowserCredential } = require("@azure/identity");
+const { KeyClient } = require("@azure/keyvault-keys");
+
+async function main() {
+  const credential = new InteractiveBrowserCredential({
+    // By default, tenantId will be "organizations". You might assign a specific tenant this way.
+    tenantId: process.env.AZURE_TENANT_ID,
+    // By default, clientId will be the same used by the Azure CLI. You might assign a specific client ID this way.
+    clientId: process.env.AZURE_CLIENT_ID  
+  });
+
+  const keyVaultUrl = `https://key-vault-name.vault.azure.net`;
+  const client = new KeyClient(keyVaultUrl, credential);
+
+  // Retrieving the properties of the existing keys in that specific Key Vault.
+  console.log(await client.listPropertiesOfKeys().next());
+}
+
+main().then(console.log).catch((e) => console.error(e));
+```
+
+As code uses the `SecretClient` in the above sample, the `InteractiveBrowserCredential` will automatically authenticate the user by launching the default system browser prompting the user to login. In this case the user interaction happens on demand as is necessary to authenticate calls from the client.
+
+## Authenticating users with the DeviceCodeCredential
+
+For terminal clients without an available web browser, or clients with limited UI capabilities the `DeviceCodeCredential` provides the ability to authenticate any client using a device code. The next sample shows authenticating a `BlobServiceClient` (from [@azure/storage-blob](https://www.npmjs.com/package/@azure/storage-blob)) using the `DeviceCodeCredential`.
+
+```ts
+const { DeviceCodeCredential } = require("@azure/identity");
+const { BlobServiceClient } = require("@azure/storage-blob");
+
+async function main() {
+  const credential = new DeviceCodeCredential(
+    // By default, tenantId will be "organizations". You might assign a specific tenant this way.
+    process.env.AZURE_TENANT_ID,
+    // By default, clientId will be the same used by the Azure CLI. You might assign a specific client ID this way.
+    process.env.AZURE_CLIENT_ID
+  );
+  const client = new BlobServiceClient("https://myaccount.blob.core.windows.net/mycontainer/myblob", credential);
+
+  const containerClient = blobServiceClient.getContainerClient("<container-name>");
+  const createContainerResponse = await containerClient.create();
+  console.log(`Successfully created a container`, createContainerResponse.requestId);
+}
+
+main().then(console.log).catch((e) => console.error(e));
+```
+
+Similarly to the `InteractiveBrowserCredential` the `DeviceCodeCredential` will also initiate the user interaction automatically as needed. To instantiate the `DeviceCodeCredential` the application must provide a callback which is called to display the device code along with details on how to authenticate to the user. By default, the full device code message will be printed to the console.
```

Comment:
Similarly here. You can describe that a device code is presented to the user, who must then use this device code on their device. Once authenticated, the user can then begin using Azure services.